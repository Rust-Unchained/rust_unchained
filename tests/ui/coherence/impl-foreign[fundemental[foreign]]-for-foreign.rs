//@ build-pass
//@ compile-flags:--crate-name=test
//@ aux-build:coherence_lib.rs

extern crate coherence_lib as lib;
use lib::*;
use std::rc::Rc;

struct Local;
struct Local1<T>(Rc<T>);


// Allowed in Unchained Rust, no conflicts here.
impl Remote1<Box<String>> for i32 {}

impl Remote1<Box<Rc<i32>>> for f64 {}

impl<T> Remote1<Box<Rc<T>>> for f32 { }

fn main() {}
