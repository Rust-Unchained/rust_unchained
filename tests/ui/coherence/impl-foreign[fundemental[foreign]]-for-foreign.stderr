error[E0117]: only traits defined in the current crate can be implemented for primitive types
  --> $DIR/impl-foreign[fundemental[foreign]]-for-foreign.rs:11:1
   |
LL | impl Remote1<Box<String>> for i32 {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^---
   | |                             |
   | |                             `i32` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: impl doesn't have any local type before any uncovered type parameters
   = note: for more information see https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules
   = note: define and implement a trait or new type instead

error[E0117]: only traits defined in the current crate can be implemented for primitive types
  --> $DIR/impl-foreign[fundemental[foreign]]-for-foreign.rs:16:1
   |
LL | impl Remote1<Box<Rc<i32>>> for f64 {
   | ^^^^^---------------------^^^^^---
   | |    |                         |
   | |    |                         `f64` is not defined in the current crate
   | |    `i32` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: impl doesn't have any local type before any uncovered type parameters
   = note: for more information see https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules
   = note: define and implement a trait or new type instead

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0117`.
